### ISSUE TITLE
Document how to detect a "detached head"

### ISSUE THREAD

**Issue by krlmlr** [r-lib/gert#139]
The code suggests checking `git_info()$shorthand`: https://github.com/r-lib/gert/blob/cdd6d399027b84bfd597d7cd5e4e188903570809/R/fetch.R#L198-L200 . Does `git_branch()` also work?

**Comment by jeroen** [comment]
Indeed, I think if the branch or the shorthand refers to `HEAD` rather than linking HEAD to a particular branch name, then it is detached.
